### PR TITLE
Fix life times of host info

### LINKED ISSUE #74: Add option to configure garbage collection on VM shutdown.
Please add an option to configure garbage collection on VM shutdown at the API level, separately from `#ifdef DEBUG`.

Currently, garbage collection in V8 is forced to run in `~StoreImpl()` only when built with `-DDEBUG`, and leaks memory otherwise. Ideally, it should always run, to avoid any memory leaks, but I understand that this is waste of CPU cycles if the process is going to be terminated anyway.

This could be exposed in the API as:
```
Store(bool fast_shutdown = false);
```

cc @jakobkummerow @rossberg

### PR DESCRIPTION
**Pull request by rossberg**
Use V8's Managed values to represent finalized host data. This way, it gets properly freed when the associated store is deleted.

Fixes #74.

@PiotrSikora, FYI.

### CODE REVIEW

File: src/wasm-v8.cc [hunk 1]
```diff
@@ -1121,33 +1119,12 @@ class RefImpl : public v8::Persistent<v8::Object> {
     // V8 attaches finalizers to handles instead of objects, and such handles
```
Comment (jakobkummerow) [inline src/wasm-v8.cc]: outdated comment

**Comment by jakobkummerow** [comment]
Note that this does not resolve the issue where someone could call `set_host_info` on a `Func` and clobber internal metadata.

File: src/wasm-v8.cc (hunk 1 above)
Comment (rossberg) [reply src/wasm-v8.cc]: Done.

**Comment by rossberg** [comment]
Created #100 to track that issue.

**Pull request merged**